packages/typescript-group-checker/src/compilers/compiler-with-watch.ts

```diff
@@ -0,0 +1,154 @@
+import { Logger } from '@stryker-mutator/api/logging';
+
+import { DependencyFile, TypescriptCompiler } from '../compiler';
+
+import { MemoryFileSystem } from '../fs/memory-filesystem';
+import { determineBuildModeEnabled, guardTSVersion, overrideOptions, retrieveReferencedProjects, toPosixFileName } from '../fs/tsconfig-helpers';
+import * as pluginTokens from '../plugin-tokens';
+
+export class CompilerWithWatch implements TypescriptCompiler {
+  public static inject = tokens(commonTokens.logger, commonTokens.options, pluginTokens.mfs);
+
+  private readonly allTSConfigFiles: Set<string>;
+  private readonly tsconfigFile: string;
+  private currentTask = new Task();
+  private currentErrors: ts.Diagnostic[] = [];
+  private readonly sourceFiles: DependencyFile[] = [];
+
+  constructor(private readonly log: Logger, private readonly options: StrykerOptions, private readonly fs: MemoryFileSystem) {
+    this.tsconfigFile = toPosixFileName(this.options.tsconfigFile);
+    this.allTSConfigFiles = new Set([path.resolve(this.tsconfigFile)]);
+  }
+
+  public async init(): Promise<{ dependencyFiles: DependencyFile[]; errors: ts.Diagnostic[] }> {
+    guardTSVersion();
+    this.guardTSConfigFileExists();
+    const buildModeEnabled = determineBuildModeEnabled(this.tsconfigFile);
+
+    const host = ts.createSolutionBuilderWithWatchHost(
+      {
+        ...ts.sys,
+        readFile: (fileName) => {
+          const content = this.fs.getFile(fileName)?.content;
+          if (content && this.allTSConfigFiles.has(path.resolve(fileName))) {
+            return this.adjustTSConfigFile(fileName, content, buildModeEnabled);
+          }
+          return content;
+        },
+        getModifiedTime: (pathName: string) => {
+          return this.fs.getFile(pathName)?.modifiedTime;
+        },
+        watchFile: (filePath: string, callback: ts.FileWatcherCallback) => {
+          const file = this.fs.getFile(filePath);
+          if (file) {
+            file.watcher = callback;
+          }
+
+          return {
+            close: () => {
+              delete this.fs.getFile(filePath)!.watcher;
+            },
+          };
+        },
+        writeFile: (fileName, data) => {
+          this.fs.writeFile(fileName, data);
+        },
+      },
+      undefined,
+      (error) => {
+        this.currentErrors.push(error);
+      },
+      (status) => {
+        this.log.debug(status.messageText.toString());
+      },
+      (summary) => {
+        this.currentTask.resolve();
+      }
+    );
+
+    host.afterProgramEmitAndDiagnostics = (program) => {
+      const sourceFiles = program.getSourceFiles().map((f: any) => {
+        const file: { fileName: string; imports: Array<{ text: string }> } = f;
+
+        const result = {
+          fileName: this.resolveFilename(file.fileName),
+          imports: new Set(
+            file.imports
+              .filter((i) => {
```

Comment:
Dit kan waarschijnlijk `filter((i) => i.text)` of  `filter((i) => Boolean(i.text))` worden